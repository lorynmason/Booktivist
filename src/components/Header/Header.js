import React from 'react';

export const Header = () => {
  return (
    <header>
      <div className="title-search">
        <h1>Booktivist</h1>
        <h3>Read up, so you can Speak up</h3>
        <input placeholder="Search for author or book"></input>
      </div>
      <img src='https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Fist.svg/2000px-Fist.svg.png' alt='activist fist'/>
      {/* <img src='../../styles/images/2000px-Fist.svg.png'/> */}
    </header>
  )
}