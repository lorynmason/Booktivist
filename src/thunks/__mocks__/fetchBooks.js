export const fetchBooks = jest.fn();