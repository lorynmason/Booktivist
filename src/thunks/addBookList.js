import { addToBookList, addMessage } from '../actions'

export const addBookList = (book) => {
  return dispatch => {
    const oldbookList = JSON.parse(localStorage.getItem('bookList'))
    if(oldbookList) {
      const updatedList = [...oldbookList, book]
      localStorage.setItem('bookList', JSON.stringify(updatedList))
      dispatch(addToBookList(updatedList))
      dispatch(addMessage(`${book.Name} was Added to Your Must Read List`))
    } else {
      localStorage.setItem('bookList', JSON.stringify([book]))
      dispatch(addToBookList([book]))
      dispatch(addMessage(`${book.Name} was Added to Your Must Read List`))
    }
  }
}