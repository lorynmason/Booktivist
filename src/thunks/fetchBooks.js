import { key } from '../apikey';
import { addSearchResults, addSearchInfo, addMessage } from '../actions'


export const fetchBooks = (search) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?k=${key}&q=${search}&type=books&verbose=1&wTeaser=item description`)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      const results = await response.json()
      dispatch(addSearchResults(results.Similar.Results))
      dispatch(addSearchInfo(results.Similar.Info))
      dispatch(addMessage(`Found Similar Books!`))
    } catch (err) {
      dispatch(addMessage(err.message))
    }
  }
}