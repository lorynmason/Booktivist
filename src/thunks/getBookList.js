import { retrieveBookList } from '../actions'

export const getBookList = () => {
  return dispatch => {
    const bookList = JSON.parse(localStorage.getItem('bookList'))
    dispatch(retrieveBookList(bookList))
  }
}