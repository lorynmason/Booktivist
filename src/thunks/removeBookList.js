import { removeBookList } from '../actions'

export const addBookList = (book) => {
  return dispatch => {
    const oldbookList = JSON.parse(localStorage.getItem('bookList'))
    const updatedList = oldbookList.filter(book => book.Name !== result.Name)
    localStorage.setItem('bookList', JSON.stringify(updatedList))
    dispatch(removeBookList(updatedList))
  }
}