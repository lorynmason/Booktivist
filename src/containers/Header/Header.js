import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBooks } from '../../thunks/fetchBooks'

export class Header extends Component {
  constructor(){
    super()
    this.state = {
      search: ''
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.props.sendSearch(this.state.search)
  }

  handleChange = (e) => {
    const { value, name } = e.target
    this.setState({
      [name]: value
    })
  }

  render() {
    return (
      <header>
        <div className="title-search">
          <h1>Booktivist</h1>
          <h3>Read up, so you can Speak up</h3>
          <form onSubmit={this.handleSubmit}>
            <input placeholder="Search for Authors or Books" name="search" value={this.search} onChange={this.handleChange}></input>
            <button>Search</button>
          </form>
        </div>
        <img src='https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Fist.svg/2000px-Fist.svg.png' alt='activist fist'/>
        {/* <img src='../../styles/images/2000px-Fist.svg.png'/> */}
      </header>
    )
  }
}

// export const mapStateToProps = state => ({

// });

export const mapDispatchToProps = dispatch => ({
  sendSearch: search => dispatch(fetchBooks(search))
});

export default connect(
  null,
  mapDispatchToProps
)(Header);