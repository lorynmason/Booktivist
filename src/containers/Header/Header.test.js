import React from 'react';
import { Header, mapDispatchToProps} from './Header';
import { shallow } from 'enzyme'
import { fetchBooks } from '../../thunks/fetchBooks'

jest.mock('../../thunks/fetchBooks');

describe('Header', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Header location={{pathname: '/'}} sendSearch={jest.fn()}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should set state of header true on componentDidMount if pathname !== "/"', () => {
    wrapper = shallow(<Header location={{pathname: '/SearchResults'}} sendSearch={jest.fn()}/>)
    const expected = {search: '', header: true}
    expect(wrapper.state()).toEqual(expected)
  })

  it('in handle change it should update state on change', () => {
    let wrapper = shallow(<Header />, { disableLifecycleMethods: true });
    wrapper.find('input').simulate('change', {
      target: {
        value: 'Into the Wild',
        name: 'search'
      }
    });
    const expected = {
      search: 'Into the Wild',
      header: false
    };
    expect(wrapper.state()).toEqual(expected);
  });

  it('in handleSubmit, if search it call sendSearch', () => {
    const mockSendSearch = jest.fn();
    wrapper = shallow(<Header sendSearch={mockSendSearch}/>, {disableLifecycleMethods: true})
    wrapper.setState({search: 'Into the Wild'})
    wrapper.handleSubmit = jest.fn();
    wrapper.sendSearch = mockSendSearch;

    wrapper.find('form').simulate('submit', {
      preventDefault: () => wrapper.handleSubmit()
    });

    expect(wrapper.handleSubmit).toHaveBeenCalled()
    expect(wrapper.sendSearch).toHaveBeenCalled();
  })

  it('mapDispatchToProps, should dispatch fetchBooks thunk when sendSearch is called from props', () => {
    const mockDispatch = jest.fn();
    fetchBooks.mockImplementation(() => {})
    const expected = fetchBooks();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.sendSearch();
    expect(mockDispatch).toHaveBeenCalledWith(expected)
  })
})