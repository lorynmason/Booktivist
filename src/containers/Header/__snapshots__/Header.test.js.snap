// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    location={
      Object {
        "pathname": "/",
      }
    }
    sendSearch={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="header-link"
          onClick={[Function]}
          replace={false}
          to="/MustReadList"
        >
          Must Read List
        </Link>,
        <header>
          <div
            className="title-search"
          >
            <h1>
              Booktivist
            </h1>
            <h3>
              Read up, so You can Speak up
            </h3>
            <form
              onSubmit={[Function]}
            >
              <input
                name="search"
                onChange={[Function]}
                placeholder="Search for Authors or Books"
                value=""
              />
              <button
                onClick={[Function]}
              >
                <Link
                  replace={false}
                  to="/SearchResults"
                >
                  Search
                </Link>
              </button>
            </form>
          </div>
          <img
            alt="activist fist"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Fist.svg/2000px-Fist.svg.png"
          />
        </header>,
      ],
      "className": "header-false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Must Read List",
          "className": "header-link",
          "onClick": [Function],
          "replace": false,
          "to": "/MustReadList",
        },
        "ref": null,
        "rendered": "Must Read List",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title-search"
            >
              <h1>
                Booktivist
              </h1>
              <h3>
                Read up, so You can Speak up
              </h3>
              <form
                onSubmit={[Function]}
              >
                <input
                  name="search"
                  onChange={[Function]}
                  placeholder="Search for Authors or Books"
                  value=""
                />
                <button
                  onClick={[Function]}
                >
                  <Link
                    replace={false}
                    to="/SearchResults"
                  >
                    Search
                  </Link>
                </button>
              </form>
            </div>,
            <img
              alt="activist fist"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Fist.svg/2000px-Fist.svg.png"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Booktivist
                </h1>,
                <h3>
                  Read up, so You can Speak up
                </h3>,
                <form
                  onSubmit={[Function]}
                >
                  <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search for Authors or Books"
                    value=""
                  />
                  <button
                    onClick={[Function]}
                  >
                    <Link
                      replace={false}
                      to="/SearchResults"
                    >
                      Search
                    </Link>
                  </button>
                </form>,
              ],
              "className": "title-search",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Booktivist",
                },
                "ref": null,
                "rendered": "Booktivist",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Read up, so You can Speak up",
                },
                "ref": null,
                "rendered": "Read up, so You can Speak up",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      name="search"
                      onChange={[Function]}
                      placeholder="Search for Authors or Books"
                      value=""
                    />,
                    <button
                      onClick={[Function]}
                    >
                      <Link
                        replace={false}
                        to="/SearchResults"
                      >
                        Search
                      </Link>
                    </button>,
                  ],
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "search",
                      "onChange": [Function],
                      "placeholder": "Search for Authors or Books",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/SearchResults"
                      >
                        Search
                      </Link>,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Search",
                        "replace": false,
                        "to": "/SearchResults",
                      },
                      "ref": null,
                      "rendered": "Search",
                      "type": [Function],
                    },
                    "type": "button",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "activist fist",
              "src": "https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Fist.svg/2000px-Fist.svg.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "header",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="header-link"
            onClick={[Function]}
            replace={false}
            to="/MustReadList"
          >
            Must Read List
          </Link>,
          <header>
            <div
              className="title-search"
            >
              <h1>
                Booktivist
              </h1>
              <h3>
                Read up, so You can Speak up
              </h3>
              <form
                onSubmit={[Function]}
              >
                <input
                  name="search"
                  onChange={[Function]}
                  placeholder="Search for Authors or Books"
                  value=""
                />
                <button
                  onClick={[Function]}
                >
                  <Link
                    replace={false}
                    to="/SearchResults"
                  >
                    Search
                  </Link>
                </button>
              </form>
            </div>
            <img
              alt="activist fist"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Fist.svg/2000px-Fist.svg.png"
            />
          </header>,
        ],
        "className": "header-false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Must Read List",
            "className": "header-link",
            "onClick": [Function],
            "replace": false,
            "to": "/MustReadList",
          },
          "ref": null,
          "rendered": "Must Read List",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title-search"
              >
                <h1>
                  Booktivist
                </h1>
                <h3>
                  Read up, so You can Speak up
                </h3>
                <form
                  onSubmit={[Function]}
                >
                  <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search for Authors or Books"
                    value=""
                  />
                  <button
                    onClick={[Function]}
                  >
                    <Link
                      replace={false}
                      to="/SearchResults"
                    >
                      Search
                    </Link>
                  </button>
                </form>
              </div>,
              <img
                alt="activist fist"
                src="https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Fist.svg/2000px-Fist.svg.png"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Booktivist
                  </h1>,
                  <h3>
                    Read up, so You can Speak up
                  </h3>,
                  <form
                    onSubmit={[Function]}
                  >
                    <input
                      name="search"
                      onChange={[Function]}
                      placeholder="Search for Authors or Books"
                      value=""
                    />
                    <button
                      onClick={[Function]}
                    >
                      <Link
                        replace={false}
                        to="/SearchResults"
                      >
                        Search
                      </Link>
                    </button>
                  </form>,
                ],
                "className": "title-search",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Booktivist",
                  },
                  "ref": null,
                  "rendered": "Booktivist",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Read up, so You can Speak up",
                  },
                  "ref": null,
                  "rendered": "Read up, so You can Speak up",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        name="search"
                        onChange={[Function]}
                        placeholder="Search for Authors or Books"
                        value=""
                      />,
                      <button
                        onClick={[Function]}
                      >
                        <Link
                          replace={false}
                          to="/SearchResults"
                        >
                          Search
                        </Link>
                      </button>,
                    ],
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "name": "search",
                        "onChange": [Function],
                        "placeholder": "Search for Authors or Books",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          replace={false}
                          to="/SearchResults"
                        >
                          Search
                        </Link>,
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Search",
                          "replace": false,
                          "to": "/SearchResults",
                        },
                        "ref": null,
                        "rendered": "Search",
                        "type": [Function],
                      },
                      "type": "button",
                    },
                  ],
                  "type": "form",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "activist fist",
                "src": "https://upload.wikimedia.org/wikipedia/commons/thumb/2/2a/Fist.svg/2000px-Fist.svg.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "header",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
